### DISCUSSION TITLE
Future of the demo store, examples and extensions

### DISCUSSION THREAD

**Discussion by elkmod** [shopware/frontends#38] [Ideas]
Shopware Frontends is built with the intention of providing a [common, low-level API](https://shopware-frontends-docs.vercel.app/packages/composables.html) that provides

* view logic
* state
* mutations

that can be used to build custom eCommerce projects with Shopware's APIs based on Vue.js.

We believe this is the right approach provides the right degree of flexibility for custom projects. Still, we provide a [boilerplate implementation](https://shopware-frontends-docs.vercel.app/getting-started/templates/demo-store-template.html) that can be seen as a "demo" for Shopware Frontends. But for now it really is not more than that - it lacks many important features, such as:

* Multi domain/currency/language support
* Synchronising configurations for checkout/registration/newsletter and more
* Snippet handling

Our current idea is to leave it like that, because adding new, more complex features to the demo store also means we have to spend more time to maintain it. We feel that kind of work is not what we should be spending a lot of effort on. This leads to the argument of extensibility. In custom projects, there's always a decision to use third party code or to build something custom - for example

* Analytics integrations
* Cookie banners
* Product configurators
* Language switches
* Payment integrations

For many of these "features" it makes sense to provide a common ground. But it is difficult to define where this "common ground" starts when our [public API](https://shopware-frontends-docs.vercel.app/packages/composables.html) is just the set of composables. There is no direct extension surface provided by the composables - as opposed to our [default Shopware storefront](https://developer.shopware.com/docs/concepts/framework/architecture/storefront-concept) where developers can build fully integrated plugins that extend everything from business logic through templates.

Having such an extension surface will drastically increase the complexity of the entire framework AND project development. Hence, we think we should follow a different approach than the default Shopware storefront. Instead, we want to publish new features in the [examples section](https://shopware-frontends-docs.vercel.app/examples/) of our documentation and enhance them with guides / descriptions. That way we can provide a new feature as a cookbook / guide with instructions on how to integrate it, but without hard-coding any dependencies / extensions points into any custom project. It comes at the price of not being plug-and-play - but then again - this is the whole idea of Shopware Frontends in the first place. Prioritising a lean, dedicated project over a bulky all-in-one suite.

For the demo store it will mean we won't create an extension layer/override mechanism/slots on top of the components - but we will leave it as lean and easy to comprehend as it is and focus on providing qualitative guides/examples for complex features rather than fully-integrated plugins.

Please join the discussion by sharing your thoughts, experiences.

**Comment by kasztof** [comment]
Where is the "examples section" available now? Neither https://frontends.shopware.com/examples/ nor https://frontends.shopware.com/resources/examples/ works. 

Or is the ["Building"](https://frontends.shopware.com/getting-started/) page equivalent of "examples section"? 

Thanks!

**Reply by BrocksiNet** [reply]
We moved it into building some time ago.

https://frontends.shopware.com/getting-started/page-elements/examples/

**Comment by saeid-elio** [comment]
Can I use Shopware's API with pure JavaScript and no frontend framework to build a shop? Considering that there are hundreds of frontend frameworks out there and new ones are emerging every month, I wish Shopware had provided some tutorial/instructions or boilerplate for developers who wish to start building shops without Vue experience and using Shopware's API alone. This way instead of biasing toward a small portion of frontend developers who use Vue as their framework of choice, you were favouring all frontend developers out there.

**Reply by BrocksiNet** [reply]
Hi @saeid-elio :wave: 

Thank you for your feedback. :raised_hands: 

We also try to make our packages and tools as platform-agnostic as possible. But honestly, the team has the most knowledge of Vue and Nuxt, and because of that, the demo-store template has the most features. 

We also have a React template; check it out [here](https://github.com/shopwareLabs/nextjs-shopware-starter). 
You can also use our new API client anywhere in JS-World; check out the NPM package [here](https://www.npmjs.com/package/@shopware/api-client). 

Is that maybe already helping you to get started?

All the best,
Björn
